pkg/controller/chaosscheduler/updateStatus.go

```diff
@@ -90,3 +80,24 @@ func inActiveList(cs chaosTypes.SchedulerInfo, uid types.UID) bool {
 func IsEngineFinished(j *operatorV1.ChaosEngine) bool {
 	return j.Status.EngineStatus == operatorV1.EngineStatusCompleted
 }
+
+// UpdateSchedulerStatus updates the scheduler status for the complete
+func (schedulerReconcile *reconcileScheduler) UpdateSchedulerStatus(cs *chaosTypes.SchedulerInfo, request reconcile.Request) error {
+	cs.Instance.Status.Schedule.Status = schedulerV1.StatusCompleted
+	cs.Instance.Status.Schedule.EndTime = &metav1.Time{Time: time.Now()}
```

Comment:
If we update the this here all the updates will have scheduler as completed. Like
https://github.com/litmuschaos/chaos-scheduler/pull/17/files#diff-d63eca6c572d7493a59874f1937e1a5e383a6307b6a4d68719b07d47faf2182cR27 and https://github.com/litmuschaos/chaos-scheduler/pull/17/files#diff-d63eca6c572d7493a59874f1937e1a5e383a6307b6a4d68719b07d47faf2182cR39

We don't want to complete the schedule at these places

Reply:
Modified this a little bit. I am calling this function when there is no possible schedules available in the future. So updating it's status and state to completed

Reply:
But if you view the above 2 links you will find it being updated in an unwanted situation.

Reply:
I think it is a valid case, if there is no possible schedule then we can patch it to complete